.github/workflows/scripts/update-e2e-stats.js

```diff
@@ -0,0 +1,171 @@
+const { google } = require('googleapis');
+const data = require(process.env.E2E_SUMMARY_JSON);
+
+// Extract the required information
+const { total, skipped, broken, failed } = data.statistic;
+const duration = data.time.duration;
+const currentDate = new Date();
+// Google Sheet ID and range
+const spreadsheetId = process.env.GOOGLE_SHEET_ID;
+
+// Credentials and authentication
+const client = new google.auth.JWT(
+  process.env.GOOGLE_CLIENT_EMAIL,
+  null,
+  process.env.GOOGLE_PRIVATE_KEY,
+  ['https://www.googleapis.com/auth/spreadsheets']
+);
+
```

Comment:
As with the other PR, there is some duplication in code here.  Just leaving it this way for now to ease updating, will refactor in the future to clean up the code.